spec/models/youtube_profile_spec.rb

```diff
@@ -7,8 +7,23 @@
   it { is_expected.to validate_presence_of :channel_id }
   it { is_expected.to have_one(:rss_feed).dependent :destroy }
   it { is_expected.to have_and_belong_to_many :affiliates }
-  it { is_expected.to validate_uniqueness_of(:channel_id).
-                with_message(/has already been added/) }
+  
+  it do
+    is_expected.to validate_uniqueness_of(:channel_id).
+      with_message(/has already been added/)
+  end
+
+  describe 'Gets the active YoutubeProfiles' do
+    let (:profiles) { YoutubeProfile.active }
+
+    it 'gets the active youtube profiles' do
+      expect(YoutubeProfile.active.count).to equal(2)
+    end
+
+    it 'is expected to have uniq values' do
+      expect(YoutubeProfile.active.count).to equal(YoutubeProfile.active.distinct.count)
+    end
+  end
```

Comment:
A few spec changes:
- use the scope method name as the description: `describe '.active' do`
- It would be preferable to explicitly create a duplicate profile to confirm the uniqueness, rather than rely on a fixture. Example:
```ruby
    context 'when the same profile belongs to multiple affiliates' do
      let(:profile) { youtube_profiles(:nps) }

      before do
        profile.affiliates << affiliates(:another_affiliate)
      end

      it 'returns unique Youtube channels' ...etc
```
That adds some extra documentation and helps keep our fixtures free of data required for a single spec.
- Add a separate spec for the behavior `it 'does not include profiles that do not belong to an affiliate'`. No spec currently tests the omission of the `noaffiliate_channel_id` profile; the specs will still pass if that fixture does not exist. Example:
```ruby
    it 'does not include profiles that do not belong to an affiliate' do
      expect(YoutubeProfile.active).not_to include(youtube_profiles(:no_affiliate))
    end
```
  

- add a spec that verifies the bugfix. `YoutubeProfile.active` on its own works fine in production. The failure only arises when that scope is combined with the `stale` scope:
```ruby
irb(main):022:0> YoutubeProfile.active.count
=> 898
irb(main):023:0> YoutubeProfile.active.stale.count
Traceback (most recent call last):
        1: from (irb):23
NoMethodError (undefined method `stale' for #<Array:0x0000556a5cbef188>)
```
Apparently the `uniq` is forcing the scope to return an array rather than a relation, which prevents it from being combined with other scopes. The simplest way to confirm the desired behavior is by verifying the class returned by the scope: 
```ruby
    it 'returns a relation' do
      expect(YoutubeProfile.active).to be_an(ActiveRecord::Relation)
    end
```

Reply:
Currently the bug is that it is using uniq rather than distinct. I agree this code does need a lot more rspec tests than it currently has. (did not have any for the scopes). But due to limited time, and other higher priority tickets, I am only adding minimum rspec test just to test out using distinct instead of uniq bug fix. We can create a ticket to address additional tests later.

Reply:
Also uniq is deprecated and removed in rails 5.0 so the test of using distinct is in this rspec test.

Reply:
https://riptutorial.com/ruby-on-rails/example/11652/-distinct--or--uniq-

Reply:
https://blog.driftingruby.com/ruby-on-rails-5-1-0-deprecations/